### PR TITLE
Adding some additional functionality to make it easier to set up Bandit from pre-existing data

### PR DESCRIPTION
**Pull request by jonathan-fulton**
Adding Arm.rewardMultiple to enable adding multiple counts / total reward to an arm

Adding a new form of constructor so you don't even have to call reward, you can simply initialize all the arms by passing in an array of {count, sum} objects

My specific use case:  running a multi-armed bandit test online for ad CTR.  I have views / clicks in the database and don't want to have to loop over reward(0)/reward(1) to get the bandit in the correct state.

### CODE REVIEW

**Comment by omphalos** [comment]
Thanks for making these changes!  I have a few minor comments.

File: bayesian-bandit.js [hunk 1]
```diff
@@ -6,7 +6,10 @@
   // Arm - one of the multi-armed bandit's arms, tracking observed rewards. //
   ////////////////////////////////////////////////////////////////////////////
 
-  function Arm() { }
+  function Arm(args) {
+    this.count = args && args.count || 0;
+    this.sum = args && args.sum || 0;
```
Comment (omphalos) [inline bayesian-bandit.js]: Now that we have this maybe we can remove the `Arm.prototype.count = 0` and `Arm.prototype.sum = 0` lines?

File: package.json [hunk 2]
```diff
@@ -25,5 +28,8 @@
   "engine": {
     "node": ">=0.8.17"
   },
-  "author": "omphalos"
+  "author": "omphalos",
+  "dependencies": {
+    "nodeunit": "^0.9.1"
+  }
```
Comment (omphalos) [inline package.json:34]: This definitely should be part of the package.json.  Would you mind making this a dev dependency though?

File: package.json [hunk 3]
```diff
@@ -7,6 +7,9 @@
     "type": "git",
     "url": "https://github.com/omphalos/bayesian-bandit.js.git"
   },
+  "scripts": {
+    "test": "NODE_ENV=test nodeunit tests.js"
```
Comment (omphalos) [inline package.json]: Does `NODE_ENV=test` add anything or can we delete it?

File: tests.js [hunk 4]
```diff
@@ -138,6 +150,21 @@ module.exports = testCase({
...
+
+      var bandit = new Bandit([{count: 100, sum: 20},{count: 50, sum: 30},{count:70, sum: 10}])
+
+      test.equal(bandit.arms.length, 3)
+      test.equal(bandit.arms[0].count, 100);
+      test.equal(bandit.arms[0].sum, 20);
+      test.equal(bandit.arms[1].count, 50);
+      test.equal(bandit.arms[1].sum, 30);
+      test.equal(bandit.arms[2].count, 70);
+      test.equal(bandit.arms[2].sum, 10);
```
Comment (omphalos) [inline tests.js]: Sorry I realize this style is a little strange to some eyes but would you mind removing the semicolons (here and elsewhere) to make it consistent with the rest of the code?

File: tests.js [hunk 5]
```diff
@@ -138,6 +150,21 @@ module.exports = testCase({
       test.done()
     },
 
+    'new Bandit with array should create the arms initialized properly': function(test) {
+
+      var bandit = new Bandit([{count: 100, sum: 20},{count: 50, sum: 30},{count:70, sum: 10}])
```
Comment (omphalos) [inline tests.js]: Another minor stylistic thing but is there a way to keep the lines under 80 chars?

File: bayesian-bandit.js [hunk 6]
```diff
@@ -70,8 +78,14 @@
...
-    for(var a = 0; a < (options || {}).numberOfArms; a++) {
-      this.arms.push(new Arm())
+    if (!Array.isArray(options)) {
+      for (var a = 0; a < (options || {}).numberOfArms; a++) {
+        this.arms.push(new Arm())
+      }
+    } else {
+      for (var i = 0; i < options.length; i++) {
+        this.arms.push(new Arm(options[i]))
+      }
```
Comment (omphalos) [inline bayesian-bandit.js]: Maybe we could accept a parameter called `arms` which is an array?  I think it makes the constructor invocation a little easier to read if the optional parameter is labeled.  What do you think?

File: package.json (hunk 2 above)
Comment (omphalos) [inline package.json:34]: I think we can also change the reference to `npm install -g nodeunit` in the README to `npm install`.

**Comment by omphalos** [comment]
Also, it might be worth adding your new constructor invocation as a section in the README.

File: .gitignore [hunk 7]
```diff
@@ -0,0 +1,2 @@
+node_modules
+.idea
```
Comment (omphalos) [inline .gitignore]: I think `.idea` probably can be moved to your own global .gitignore.

I agree `node_modules` should definitely be in here though.

**Comment by omphalos** [comment]
@jonathan-fulton those are all the comments I have.  Thanks again for making these changes.  If you think I'm off-base about anything let me know.  I appreciate the work you've done and I think it's a nice improvement that could potentially help other users of the library as well.

File: bayesian-bandit.js (hunk 1 above)
Comment (jonathan-fulton) [inline bayesian-bandit.js]: Good call.  Will update.

File: package.json (hunk 2 above)
Comment (jonathan-fulton) [inline package.json:34]: Good call on both points, let me change.

File: package.json (hunk 3 above)
Comment (jonathan-fulton) [inline package.json]: For most of my projects I run tests under the "test" environment.  It doesn't really add anything here though so I'll go ahead and remove.  It mostly matters when there's differences b/w prod / test / local environments and we don't have anything like that here.

File: tests.js (hunk 4 above)
Comment (jonathan-fulton) [inline tests.js]: Yep!

File: tests.js (hunk 5 above)
Comment (jonathan-fulton) [inline tests.js]: Done

File: bayesian-bandit.js (hunk 6 above)
Comment (jonathan-fulton) [inline bayesian-bandit.js]: Yea, great point. Let me update it.

File: .gitignore (hunk 7 above)
Comment (jonathan-fulton) [inline .gitignore]: Yea, good point.

**Comment by jonathan-fulton** [comment]
@omphalos Thanks for the great comments (and creating the project in the first place)!  I'm about to push all the changes.  I also decided to add JSDoc comments while I was in there so let me know if those are ok.

**Comment by omphalos** [comment]
Ok - looks good.  The JSDoc comments are a nice touch.  Thanks again!

**Comment by jonathan-fulton** [comment]
Of course!

**Pull request merged**